pkg/sql/conn_executor.go

```diff
@@ -3113,6 +3112,79 @@ func (ex *connExecutor) handleWaitingForDescriptorIDGeneratorMigration(ctx conte
 	return ex.resetTransactionOnSchemaChangeRetry(ctx)
 }
 
+func (ex *connExecutor) waitOneVersionForNewVersionDescriptorsWithoutJobs(
```

Comment:
this file is too big already. What do you think about moving all this job related code to a new file `conn_executor_jobs.go` and moving the `jobsCollection` and its friends to a file `jobs_collection.go`?

Reply:
good point. moved them to new files.